rust-wasm-id-allocator/distributed-id-allocator/src/compressor/tables/session_space.rs

```diff
@@ -10,13 +11,18 @@ use std::ops::Bound;
 pub struct Sessions {
     session_map: BTreeMap<SessionId, SessionSpaceRef>,
     session_list: Vec<SessionSpace>,
+    // Session IDs are stored in little endian byte form in a compact array to accelerate serialization.
+    // When a session ID is queried (via session space ref) the bytes are converted to a u128/Session ID.
+    // This tradeoff favors serialization at a small penalty to the (already slow) path of decompress/recompress.
+    session_ids: Vec<u8>,
 }
 
 impl Sessions {
     pub fn new() -> Sessions {
         Sessions {
             session_map: BTreeMap::new(),
             session_list: Vec::new(),
+            session_ids: Vec::new(),
```

Comment:
So is this a "parallel" vector to session_list, such that either can be referenced by the same index (via SessionSpaceRef)?

Reply:
Exactly, but indexed by index*16 (u128 byte size).
It makes it one simple memcopy to put into the byte vec during serialization (via extend_by_slice).